salt/modules/win_pkg.py

```diff
@@ -647,10 +696,12 @@ def refresh_db(**kwargs):
         )
 
     # Cache repo-ng locally
+    log.info('Fetching *.sls files from {0}'.format(repo_details.winrepo_source_dir))
     __salt__['cp.cache_dir'](
-        repo_details.winrepo_source_dir,
-        saltenv,
-        include_pat='*.sls'
+        path=repo_details.winrepo_source_dir,
+        saltenv=saltenv,
+        include_pat='*.sls',
+        exclude_pat=r'E@\/\..*\/'  # Exclude all hidden directories (.git)
```

Comment:
The pattern is probably ok in this use case, but it is a greedy match, so it's not grabbing _just_ the hidden directory but every directory in the path up to the final `/`

For example, with `salt://win/repo-ng/.git/foo/init.sls` it matches `/.git/foo/`.

To make it non-greedy, just add a `?` after the wildcard: `r'E@\/\..*?\/'`

Reply:
I agree on this one. It would likely work here because assuming that you have several dirs under the `.git` dir, the pattern would match each of them individually. It may not cause a problem doing it this way, but making it a non-greedy match will mean that we only match on `.git/` and then anything under it is automatically excluded because the parent directory is excluded.